Finacial Domain/Proximity to POI/POI.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split, GridSearchCV, cross_val_score
+from sklearn.linear_model import LinearRegression, Ridge, Lasso
+from sklearn.preprocessing import StandardScaler, PolynomialFeatures, OneHotEncoder
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.impute import SimpleImputer
+from sklearn.metrics import mean_squared_error, r2_score, mean_absolute_error
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# Load the dataset
+data_path = '/kaggle/input/derivatives-trading/pred2016l.csv'
+data = pd.read_csv(data_path)
+
+# Inspect the columns to identify the target variable
+print(data.columns)
+
+# Identify target and feature columns
+target = 'YourActualTargetColumnName'  # Replace this with the actual target column name
+features = data.columns.drop(target)
+
+# Separate numerical and categorical columns
+num_cols = data.select_dtypes(include=['float64', 'int64']).columns.drop(target)
+cat_cols = data.select_dtypes(include=['object']).columns
+
+# Define preprocessing steps for numerical and categorical data
+num_transformer = Pipeline(steps=[
+    ('imputer', SimpleImputer(strategy='median')),
+    ('scaler', StandardScaler())
+])
+
+cat_transformer = Pipeline(steps=[
+    ('imputer', SimpleImputer(strategy='most_frequent')),
+    ('onehot', OneHotEncoder(handle_unknown='ignore'))
+])
+
+# Combine preprocessing steps
+preprocessor = ColumnTransformer(
+    transformers=[
+        ('num', num_transformer, num_cols),
+        ('cat', cat_transformer, cat_cols)
+    ])
+
+# Define the full pipeline including polynomial features and the regression model
+pipeline = Pipeline(steps=[
+    ('preprocessor', preprocessor),
+    ('poly_features', PolynomialFeatures(degree=2, include_bias=False)),
```

Comment:
Remove 'degree=2' to be set in GridSearchCV

```suggestion
    ('poly_features', PolynomialFeatures( include_bias=False)),
```